aptos/typescript/src/gas_station.ts

```diff
@@ -0,0 +1,188 @@
+
+// Create Aptos client for building and submitting our transactions.
+const aptos = new Aptos(new AptosConfig({ network: Network.TESTNET}));
+
+// Create Shinami Gas Station client for sponsoring our transactions.
+const SHINAMI_TESTNET_GAS_KEY = "{{APTOS_TESTNET_GAS_STATION_ACCESS_KEY}}";
+const gasStationClient = new GasStationClient(SHINAMI_TESTNET_GAS_KEY);
+
+// let executedTransaction = await buildSponsorSignAndSubmitSimpleTransaction();
+let executedTransaction = await buildSponsorSignAndSubmitMultiAgentTransaction();
+console.log("\nTransaction hash:", executedTransaction.hash);
+console.log("Transaction status:", executedTransaction.vm_status);
+
+
+
+// Build, sponsor, sign, and execute a simple Move call transaction
+async function buildSponsorSignAndSubmitSimpleTransaction(): Promise<CommittedTransactionResponse> {
+    // 1. Set up our sender.
+    //    Replace function call with private key value printed to console after first run.
+    const PRIVATE_KEY = await generateSingleKeyAccountPrivateKey(); 
+    console.log("\nSender's private key:", PRIVATE_KEY);
+
+    const sender: SingleKeyAccount = new SingleKeyAccount({ privateKey: new Ed25519PrivateKey(PRIVATE_KEY) }); 
+    console.log("Sender address:", sender.accountAddress.toString());
+    
+    // 2. Build a simple transaction.
+    let transaction = await buildSimpleMoveCallTransaction(sender.accountAddress);
+
+    // 3. Sponsor the transaction with Shinami Gas Station.
+    let feePayerAuthenticator = await gasStationClient.sponsorTransaction(transaction);
+    // Note that the SDK updates the transaction's feePayer address on a successful sponsorship
+    console.log("\ntransaction.feePayerAddress post-sponsorship:", transaction.feePayerAddress);
+    
+    // 4. Generate the sender's signature. 
+    const senderAuthenticator = aptos.transaction.sign({ 
+        signer: sender, 
+        transaction 
+    });
+    
+    // 5. Submit the transaction with the sender and fee payer signatures
+    const committedTransaction = await aptos.transaction.submit.simple({
+        transaction,
+        senderAuthenticator,
+        feePayerAuthenticator: feePayerAuthenticator,
+    });
+    
+    // 6. Wait for and retrieve the executed transaction
+    const executedTransaction = await aptos.waitForTransaction({
+        transactionHash: committedTransaction.hash
+    });
+    
+    return executedTransaction;
+}
+
+
+
+// Helper function to generate a private key for a SingleKeyAccount.
+//  If the argument is set to true, the account will be funded.
+async function generateSingleKeyAccountPrivateKey(fund = false) : Promise<string> {
+    const account: SingleKeyAccount = SingleKeyAccount.generate();
+    if (fund) {
+        await aptos.fundAccount({
+            accountAddress: account.accountAddress,
+            amount: 100000000
+        });
+    }
+    
+    const privateKey = Buffer.from(account.privateKey.toUint8Array()).toString('hex');
```

Comment:
I suggest directly returning the `Account` as opposed to going back and forth with the private key, which has a lot of assumptions of what type of key it is.